server/asset/btc/rpcclient.go

```diff
@@ -0,0 +1,142 @@
+	RawRequest(context.Context, string, []json.RawMessage) (json.RawMessage, error)
+	Shutdown()
+	WaitForShutdown()
+}
+
+// RPCClient is a bitcoind wallet RPC client that uses rpcclient.Client's
+// RawRequest for wallet-related calls.
+type RPCClient struct {
+	ctx       context.Context
+	requester RawRequester
+}
+
+func (rc *RPCClient) callHashGetter(method string, args anylist) (*chainhash.Hash, error) {
+	var txid string
+	err := rc.call(method, args, &txid)
+	if err != nil {
+		return nil, err
+	}
+	return chainhash.NewHashFromStr(txid)
+}
+
+// GetBestBlockHash returns the hash of the best block in the longest block
+// chain.
+func (rc *RPCClient) GetBestBlockHash() (*chainhash.Hash, error) {
+	return rc.callHashGetter(methodGetBestBlockHash, nil)
+}
+
+// GetBlockchainInfoResult models the data returned from the getblockchaininfo
+// command.
+type GetBlockchainInfoResult struct {
+	Blocks               int64  `json:"blocks"`
+	Headers              int64  `json:"headers"`
+	BestBlockHash        string `json:"bestblockhash"`
+	InitialBlockDownload bool   `json:"initialblockdownload"`
+}
+
+// GetBlockChainInfo returns information related to the processing state of
+// various chain-specific details.
+func (rc *RPCClient) GetBlockChainInfo() (*GetBlockchainInfoResult, error) {
+	chainInfo := new(GetBlockchainInfoResult)
+	err := rc.call(methodGetBlockchainInfo, nil, chainInfo)
+	if err != nil {
+		return nil, err
+	}
+	return chainInfo, nil
+}
+
+// EstimateSmartFee requests the server to estimate a fee level based on the
+// given parameters.
+func (rc *RPCClient) EstimateSmartFee(confTarget int64, mode *btcjson.EstimateSmartFeeMode) (*btcjson.EstimateSmartFeeResult, error) {
+	res := new(btcjson.EstimateSmartFeeResult)
+	return res, rc.call(methodEstimateSmartFee, anylist{confTarget, mode}, res)
+}
+
+// GetTxOut returns the transaction output info if it's unspent and
+// nil, otherwise.
+func (rc *RPCClient) GetTxOut(txHash *chainhash.Hash, index uint32, mempool bool) (*btcjson.GetTxOutResult, error) {
+	// Note that we pass to call pointer to a pointer (&res) so that
+	// json.Unmarshal can nil the pointer if the method returns the JSON null.
+	var res *btcjson.GetTxOutResult
+	return res, rc.call(methodGetTxOut, anylist{txHash.String(), index, mempool},
+		&res)
+}
+
+// GetRawTransaction retrieves tx's information.
+func (rc *RPCClient) GetRawTransaction(txHash *chainhash.Hash) (*btcutil.Tx, error) {
+	res := new(btcutil.Tx)
+	return res, rc.call(methodGetRawTransaction, anylist{txHash.String()}, res)
+}
+
+// GetRawTransactionVerbose retrieves tx's information with verbose flag set
+// to true.
+func (rc *RPCClient) GetRawTransactionVerbose(txHash *chainhash.Hash) (*btcjson.TxRawResult, error) {
+	res := new(btcjson.TxRawResult)
+	return res, rc.call(methodGetRawTransaction, anylist{txHash.String(),
+		true}, res)
+}
+
+// GetBlockVerbose returns a data structure from the server with information
+// about a block given its hash.
```

Comment:
How about "GetBlockVerbose fetches verbose block data for the specified hash." ?